sysconfigs/uenv/pre-setup.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+conf_dir="$parent_dir"/sysconfigs/uenv
+conf_files=(compilers.yaml upstreams.yaml packages.yaml)
+
+# lock the current spack instance to a specific uenv
+if [[ ! -f $uenv_mount/meta/hash ]]; then
+    echo "ERROR: file $uenv_mount/meta/hash not found, spack-c2sm cannot be configured for this uenv"
+    return 1
+fi
+uenv_hash=$(cat $uenv_mount/meta/hash)
+if [[ -f "$conf_dir"/uenv.hash ]]; then
+    local_hash=$(cat "$conf_dir"/uenv.hash)
+    if [[ $uenv_hash != $local_hash ]]; then
+        echo "ERROR: This spack-c2sm instance was already once configured with a different uenv."
+        echo "       Please reuse the same uenv with the current instance or setup a fresh"
+        echo "       spack-c2sm with any uenv."
```

Comment:
More info about required spack command for cleanup would be nice. I.e. `run spack clean -a` or so.

Reply:
Hum, I intentionally just wrote "setup a fresh sapck-c2sm" so that users would start from scratch and not try to clean the current one. I don't think they should try to clean, actually.